### ISSUE TITLE
Bug: Issue when searching using equality operator

### ISSUE THREAD

**Issue by ahmedisam99** [redis/redis-om-dotnet#469]
```csharp
await _myCollection.Where(x => x.Name == "Test").ToListAsync();
// The above line gets translated to:
// "FT.SEARCH" "idx" (@Name:"Test")
// This fails in redis-cli but returns an empty list (does not throw an exception)

await _myCollection.Where(x => x.Name.Contains("Test")).ToListAsync();
// The above line gets translated to:
// "FT.SEARCH" "idx" (@Name:Test)
```

The second example works just fine, but the first does not work, because when adding quotes to the predicate, redis throws/returns the following error:

```
"Invalid argument(s): Closing quote must be followed by a space or nothing at all."
```

PS//
This works ✅ :
`"FT.SEARCH" "idx" @Name:"Test"`

But this does not 🚫:
`"FT.SEARCH" "idx" (@Name:"Test")`

**Comment by ahmedisam99** [comment]
hi @slorello89 , I'm using `RediSearch v2.8.9`

Actually, I just found out that it worked fine and  didn't return results for me because using the equality operator just returns exact matches (i.e. the whole Name must be x), What caused this is that when I copy the executed command (from `MONITOR`) and try to execute it myself it throws the error: `"Invalid argument(s): Closing quote must be followed by a space or nothing at all.` This is still a mystery for me

**Comment by ahmedisam99** [comment]
![image](https://github.com/user-attachments/assets/d17a01bb-ec51-488a-8475-896d4bd03625)

**Comment by ahmedisam99** [comment]
Hello @slorello89, This is me again 😅

I just got the hangs over what I was missing
1. The equality operator is used for exact matches for numeric, tag, and text fields (and so it didn't apply stemming for my search term and so I got the unexpected empty results)
2. The thrown error was caused by me not escaping the quotes `"FT.SEARCH" "test-idx" "(@name:\"Testing\")"`

Very sorry for the inconvenience.
